newdle/client/src/components/TimezonePicker.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+import {Dropdown} from 'semantic-ui-react';
+import {useDispatch, useSelector} from 'react-redux';
+import {getTimezone} from '../selectors';
+import {setTimezone} from '../actions';
+
+import {commonTimezones} from '../util/timezones';
+import styles from './TimezonePicker.module.scss';
+
+export default function TimezonePicker() {
+  const dispatch = useDispatch();
+  const timezone = useSelector(getTimezone);
+  const options = commonTimezones.map(({name, caption}) => {
+    return {key: name, value: name, text: name, description: caption, active: name === timezone};
+  });
+  return (
+    <div>
+      <span>Timezone:</span>
+      <Dropdown
```

Comment:
the dropdown has really awful performance here (and slows down the page render by ~150ms which is very noticable).

Semantic-Org/Semantic-UI-React#1674 might be related

Reply:
Doesn't feel that heavy here. But there isn't much we can do.